src/parse.rs

```diff
@@ -138,32 +158,44 @@ pub fn parse(
     let mut line_buffer = vec![];
 
     while let Ok(bytes_read) = reader.read_until(0x0A, &mut line_buffer) {
-        if bytes_read <= 0 {
+        if bytes_read == 0 {
```

Comment:
Actually, clippy also called this out:

```
error: this comparison involving the minimum or maximum element for this type contains a case that is always true or always false
   --> src/parse.rs:161:12
    |
161 |         if bytes_read <= 0 {
    |            ^^^^^^^^^^^^^^^
    |
    = help: because `0` is the minimum value for this type, the case where the two sides are not equal never occurs, consider using `bytes_read == 0` instead
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#absurd_extreme_comparisons
    = note: `#[deny(clippy::absurd_extreme_comparisons)]` on by default
```

Including the change, per the above reasoning.